src/lp/fixed_variable.jl

```diff
@@ -1,20 +1,105 @@
+@doc raw"""
+    FixedVariableRule
+
+Eliminate all fixed variables.
+
+## Presolve
+
+A variable ``x_{j}`` with lower and upper bounds ``l_{j}, u_{j}``
+is fixed to its lower bound if
+```math
+| l_{j} - u_{j}| \leq ϵ,
+```
+where ``\epsilon`` is a prescribed tolerance.
+
+The variable is then eliminated from the problem as follows: ``\forall i,`` row ``i``
+```math
+l^{r}_{i} \leq a_{i, j}x_{j} + \sum_{k \neq j} a_{i, k} x_{k} \leq u^{r}_{i}
+```
+is transformed into
+```math
+l^{r}_{i} - a_{i, j} l_{j} \leq \sum_{k \neq j} a_{i, k} x_{k} \leq u^{r}_{i} - a_{i, j} l_{j}.
+```
+
+## Integer variables
+
+This rule applies to integer and continuous variables.
+
+The problem is declared infeasible
+if an integer variable ``x_{j}`` is fixed to a fractional value ``l_{j}``, i.e., if
+```math
+\min(l_{j} - ⌊ l_{j} ⌋, ⌈ l_{j} ⌉ - l_{j}) \geq \epsilon_{int},
+```
+where ``\epsilon_{int}`` is a prescribed tolerance.
+
+## Postsolve
+
+For dual variables, first compute
+```math
+z_{j} = c_{j} - \sum_{i} a_{i, j} (y_{i}^{l} - y_{i}^{u}),
+```
+then recover ``z_{j}^{l} = z_{j}^{+}`` and ``z_{j}^{u} = z_{j}^{-}``.
+
+
+## Misc
+
+* This is a primal reduction.
+* This rule does not create any fill-in.
+"""
+struct FixedVariableRule <: AbstractPresolveRule end
+
 struct FixedVariable{T} <: PresolveTransformation{T}
     j::Int  # variable index
     x::T  # primal value
     c::T  # current objective coeff
     col::Col{T}  # current column
 end
 
-function remove_fixed_variable!(ps::PresolveData, j::Int)
+function apply!(
+    ps::PresolveData{T},
+    ::FixedVariableRule,
+    config::PresolveOptions{T}
+) where {T}
+
+    for (j, flag) in enumerate(ps.colflag)
+        remove_fixed_variable!(ps, j, config.PrimalTolerance, config.IntegerTolerance)
+        ps.status == NOT_INFERRED || return ps.status
+    end
+
+    return ps.status
+end
+
+"""
+    remove_fixed_variable!(ps, j; ϵ, ϵᵢ) -> ModelStatus
+
+Remove variable `xⱼ` if fixed. See [`FixedVariableRule`](@ref)
+"""
+function remove_fixed_variable!(
+    ps::PresolveData{T},
+    j::Int,
+    ϵ::T=eps(T),
+    ϵᵢ::T=eps(T)
+) where {T}
     ps.colflag[j] || return nothing  # Column was already removed
-    
-    # Check bounds
+
+    # Get current bounds
     lb, ub = ps.lcol[j], ps.ucol[j]
 
-    # TODO: use tolerance
-    if lb == ub
+    if abs(ub - lb) <= ϵ
         @debug "Fix variable $j to $lb"
 
+        # Infeasibility check for integer variables
+        if is_integer(ps.pb0.var_types[j])
+            # Check if fixing to fractional value
+            f = min(lb - floor(lb), ceil(lb) - lb)
+
```

Comment:
Nit: extra space

Reply:
"extra space" that I should remove, or that I should add?

Reply:
Sorry, I was referring to the extra newline before the ``if`` which can be removed IMO